src/viking.js

```diff
@@ -1,11 +1,91 @@
-// Soldier
-function Soldier() {}
+// Iteration 1 - Soldier
 
-// Viking
-function Viking() {}
+function Soldier(health, strength) {
 
-// Saxon
-function Saxon() {}
+  // const soldier = {};
 
-// War
-function War() {}
+  this.health = health;
+  this.strength = strength;
+  /*this.attack = function () {
+    return this.strength;
+  }*/
+  this.receiveDamage = function (damage) {
+    this.health = this.health - damage;
+  }
+}
+Soldier.prototype.attack = function () {
+  return this.strength;
+};
+
+
+// Iteration 2 - Viking
+
+function Viking(name, health, strength) {
+
+  // const viking = {};
+
+  this.name = name;
+  this.health = health;
+  this.strength = strength;
+  
+  this.receiveDamage = function (damage) {
+    this.health = this.health - damage;
+    if (this.health > 0) {
+      return `${this.name} has received ${damage} points of damage`; 
+    }
+    if (this.health <= 0) {
+      return `${this.name} has died in act of combat`;
+    }
+  };
+  this.battleCry = function () {
+    return "Odin Owns You All!";
+  };
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking
+
+
+// Iteration 3 Saxon
+
+function Saxon (health, strength) {
+  this.health = health;
+  this.strength = strength;
+  
+  this.receiveDamage = function (damage) {
+    this.health = this.health - damage;
+    if (this.health > 0) {
+      return `A Saxon has received ${damage} points of damage`; 
+    }
+    if (this.health <= 0) {
+      return `A Saxon has died in combat`;
+    }
+  };
+}
+
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.constructor = Saxon
+
+
+// Iteration 4 - War
+function War() {
+  this.vikingArmy = [];
+  this.saxonArmy = [];
+  this.addViking = function (Viking) {
+    //this.Viking = Viking {};
+    this.vikingArmy.push(Viking);
+  };
+  this.addSaxon = function (Saxon) {
+    this.saxonArmy.push(Saxon);;
+  };
+  /*
+  this.vikingAttack = function () {
```

Comment:
here you could have created a var to pick a random soldier from the array and then made this random Saxon to recieve damage from a random Viking. Then you check if health < 0. If so, splice that Saxon from the array.